modules/service/src/test/scala/QuerySuite.scala

```diff
@@ -14,6 +14,32 @@ class QuerySuite extends OdbSuite {
           nodes {
             id
             name
+            proposal {
+              title
+              proposalClass {
+                __typename
+                minPercentTime
+                ... on LargeProgram {
+                  minPercentTotalTime
+                  totalTime {
+                    seconds
+                  }
+                } 
```

Comment:
In other parts of the API where there are multiple options (e.g., source profile point, uniform and gaussian and target sidereal and nonsidereal tracking), the pattern is to put all the fields as optional.  There are benefits and drawbacks to that approach but I've tried to make it consistent.  This would be a departure, which may be fine.  Maybe the sheer number of options makes it more desirable to do it this way?

Reply:
I struggled a bit with this. In the temporary implementation in Explore, the proposal model was flatter, with an enumeration for the Proposal Class and the `minPercentTime`, `minPercentTotalTime` and `totalTime` fields directly on the proposal. But, with the latter 2 fields only being applicable to a small subset of the proposal classes, it seemed it would be confusing as to when they were required and when they were forbidden. From the UI perspective that isn't too bad (and is probably easier), but it seemed less than ideal for API users. 
If we want to change this, I'm not irrevocably tied to this approach.

Reply:
Hmmm. Yeah. I see. OK. I'll merge and we'll ponder it as we go forward. The ability to cut and paste from output to input is kind of nice.